NetworkingLabKenny/app/src/main/java/com/korbkenny/networkinglabkenny/MainActivity.java

```diff
@@ -0,0 +1,187 @@
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.widget.Button;
+import android.widget.LinearLayout;
+import android.widget.Toast;
+
+import com.android.volley.Cache;
+import com.android.volley.Network;
+import com.android.volley.RequestQueue;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.BasicNetwork;
+import com.android.volley.toolbox.DiskBasedCache;
+import com.android.volley.toolbox.HurlStack;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.JsonRequest;
+import com.android.volley.toolbox.StringRequest;
+import com.android.volley.toolbox.Volley;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+public class MainActivity extends AppCompatActivity {
+
+    public static final String GET_CEREAL_URL = "http://api.walmartlabs.com/v1/search?query=cereal&format=json&apiKey=kshb4tez4easan9u46ue4u8s";
+    public static final String GET_CHOCOLATE_URL = "http://api.walmartlabs.com/v1/search?query=chocolate&format=json&apiKey=kshb4tez4easan9u46ue4u8s";
+    public static final String GET_TEA_URL = "http://api.walmartlabs.com/v1/search?query=tea&format=json&apiKey=kshb4tez4easan9u46ue4u8s";
+    Button mCereal, mChocolate, mTea;
+    LinearLayout mButtons;
+    RecyclerView mRecyclerView;
+    ItemsListAdapter mAdapter;
+    ArrayList<String> mList;
+    RequestQueue mQueue;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mQueue = Volley.newRequestQueue(this);
+        mList = new ArrayList<>();
+
+        Cache cache = new DiskBasedCache(getCacheDir(), 1024*1024);
+        Network network = new BasicNetwork(new HurlStack());
+        mQueue = new RequestQueue(cache,network);
+        mQueue.start();
+
+        mCereal = (Button) findViewById(R.id.cereal_button);
+        mChocolate = (Button) findViewById(R.id.chocolate_button);
+        mTea = (Button) findViewById(R.id.tea_button);
+        mRecyclerView = (RecyclerView) findViewById(R.id.recycler_view);
+
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(MainActivity.this, LinearLayoutManager.VERTICAL,false);
+        mRecyclerView.setLayoutManager(linearLayoutManager);
+        mAdapter = new ItemsListAdapter(mList);
+        mRecyclerView.setAdapter(mAdapter);
+
+        mButtons = (LinearLayout) findViewById(R.id.buttons);
+
+
+        ConnectivityManager connectivityManager = (ConnectivityManager) getSystemService(CONNECTIVITY_SERVICE);
+        NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+
+        if(networkInfo != null && networkInfo.isConnected()){
```

Comment:
It's better to check for connectivity after a button is clicked, not just once in `onCreate()`. What if the user starts the app, then turns off airplane mode, then clicks a button?